api/facades/client/leadership/leadership.go

```diff
@@ -36,22 +36,25 @@ func NewClientFromFacades(clientFacade base.ClientFacade, facade base.FacadeCall
 }
 
 // PinLeadership (leadership.Pinner) sends a request to the API server to pin
-// leadership for the input application.
-func (l *client) PinLeadership(appName string) error {
-	return errors.Trace(l.pinOp(appName, "PinLeadership"))
```

Comment:
Given the client is already connected as a specific entity, is there a could this entity be inferred by the client state?
Otherwise on server side we have to check "is this entity that is logged in able to have authority over the entity that they are making the request as".